src/main/resources/db/migration/R__functions.sql

```diff
@@ -267,21 +267,21 @@ $body$
     LANGUAGE plpgsql;
 
 
+CREATE OR REPLACE VIEW latest_version AS
+  SELECT *
+    FROM versions
+    ORDER BY id DESC
+    LIMIT 1;
+
 -- Return deltas that reasonable to put into the notification.xml file.
 -- These has to be
 -- 1) deltas from the latest session
 -- 2) of total size not larger than the latest snapshot
 -- 3) having a definite hash (latest delta can have undefined hash until its file is generated and written)
 CREATE OR REPLACE VIEW reasonable_deltas AS
-WITH latest_version AS (
-    SELECT *
-    FROM versions
-    ORDER BY id DESC
-    LIMIT 1
-)
 SELECT z.*
 FROM (SELECT v.*,
-             SUM(delta_size) OVER (ORDER BY id DESC) AS total_delta_size
+             SUM(delta_size) OVER (PARTITION BY session_id ORDER BY id DESC) AS total_delta_size
```

Comment:
Oh, that's a good catch!

Reply:
Ah wait, there's `WHERE z.session_id = latest_version.session_id` so there's only one session id.